cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java

```diff
@@ -48,35 +48,45 @@
 public final class GossipProtocol implements IGossipProtocol, IManagedGossipProtocol {
   private static final Logger LOGGER = LoggerFactory.getLogger(GossipProtocol.class);
 
-  private ITransport transport;
+  private final String memberId;
+  private final ITransport transport;
+  private final ScheduledExecutorService executor;
```

Comment:
Always wondered why ``gossip`` does have its own executor, while ``clusterMembership`` and ``failureDetector`` don't. Can't we in this PR create dedicated executors for ``clusterMembership`` and ``failureDetector`` -s ?